tests/Tests/Unit/BidiStreamTest.php

```diff
@@ -195,6 +194,10 @@ public function testReadFailure()
                 $this->assertSame($response, $responses[$index]);
                 $index++;
             }
+        } catch(ApiException $e) {
+            $this->assertStringContainsString('read failure', $e->getMessage());
+        } catch(Exception $e) {
+            $this->fail('An '.ApiException::class.' should have been thrown');
```

Comment:
Instead we should use `expectException` and `expectExceptionMessage` here. Without a catch, the exception gets thrown, so there's no need to add the catch and assertions (if the annotations work, the function call versions will work as well).

Reply:
Dangit, I was sure I was running into errors when just using the `->expectException*` methods. I'll change them, of course!